tools/qinfo.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+#
+# This program reads all logged queries in the database and dumps
+# various stats wrt. unanswered queries + more.
+
+import os
+import sys
+
+# Hack to make this Python program executable from the tools subdirectory
+basepath, _ = os.path.split(os.path.realpath(__file__))
+_TOOLS = os.sep + "tools"
+if basepath.endswith(_TOOLS):
+    basepath = basepath[0 : -len(_TOOLS)]
+    sys.path.append(basepath)
+
+from collections import Counter
+from pprint import pprint
+
+from db import SessionContext
+from db.models import QueryLog
+from queries.special import _SPECIAL_QUERIES
+
+from Levenshtein import StringMatcher
+
+
+_MAX_DISTANCE = 5
+_MIN_QUERY_LEN = 10
+
+
+def main() -> None:
+
+    uniq = set()
+    special = _SPECIAL_QUERIES.keys()
+
+    # Read all logged queries from database
+    with SessionContext(read_only=True) as session:
+        counter = Counter()
+
+        ql = session.query(QueryLog).filter(QueryLog.answer == None)
+        for q in ql:
+            question = q.question
+            # if (
+            #     len(question) < _MIN_QUERY_LEN
+            #     or (question in uniq)
+            #     or (question in special)
+            # ):
+            #     continue
+            # print(question)
+            counter[question] += 1
+            continue
```

Comment:
The code below the `continue` is never executed, is that intentional?

Reply:
Yes. This is all kind of hacky and for personal use, like many of the other tools I've added. Sometimes I want certain filtering criterions, at other times not, couldn't be bothered to implement getopt parsing etc.